Source/santa-driver/SantaDecisionManager.cc

```diff
@@ -14,9 +14,52 @@
 
 #include "SantaDecisionManager.h"
 
+// This is a made-up KAUTH_FILEOP constant which represents a
+// KAUTH_VNODE_WRITE_DATA event that gets passed to SantaDecisionManager's
+// FileOpCallback method.  The KAUTH_FILEOP_* constants are defined in
+// sys/kauth.h and run from 1--7.  KAUTH_VNODE_WRITE_DATA is already defined as
+// 4 so it overlaps with the other KAUTH_FILEOP_* constants and can't be used.
+// We define KAUTH_FILEOP_WRITE as something much greater than 7.
+#define KAUTH_FILEOP_WRITE 100
+
 #define super OSObject
 OSDefineMetaClassAndStructors(SantaDecisionManager, OSObject);
 
+# pragma mark Monitoring PIDs
+
+// This keeps track of all pids associated with compiler processes.  It is defined as a global
+// variable so that the pid monitor threads can access it without needing to reference
+// our instance of SantaDecisionManager.
+static SantaCache<bool> *compiler_pid_set_ = new SantaCache<bool>(500, 5);
```

Comment:
As discussed:
* make this a property of SDM
* add a counter to SDM (like listener_invocations_) to track the number of threads that are outstanding
* atomically inc/dec that counter before making and terminating the threads
* during SDM dealloc (or ClientDisconnect, maybe?) make this cache pointer NULL to make the threads begin terminating and then sleep-loop until the number of outstanding threads drops to 0.
* use a struct to pass the SDM reference and the pid to the threads when they start

Reply:
Done. Added pid_monitor_thread_count_ to track the number of threads.  Added PidMonitorThreadsShouldExit() function for the threads to check, since cache pointer is private.  Currently calling StopPidMonitorThreads() from SDM's free() method.  Alternately I think maybe this could be called from SantaDriver::stop() similar to StopListener.  Not sure which is better.